autoprecompiles/src/adapter.rs

```diff
@@ -22,6 +24,7 @@ pub trait Adapter: Sized {
     type Program: Program<Self::Instruction> + Send;
     type Instruction: Instruction<Self::Field> + Serialize + for<'de> Deserialize<'de> + Send;
     type MemoryBusInteraction: MemoryBusInteraction<Self::PowdrField, AlgebraicReference>;
+    type BusType: Clone + Display + Sync + Eq + PartialEq;
```

Comment:
Not sure if it's worth making this generic, or just use `String`. We only really use it to have a unique identifier and print rendering constraints. Introducing this type parameter causes most of the changes in this PR.

Any opinions?

Reply:
What would be uglier is if the integration matches on the bus type, e.g. [here](https://github.com/powdr-labs/powdr/blob/92ae7469e817d71dbf8054e5241317296f4ef968/openvm/src/bus_interaction_handler/mod.rs#L65). There, we would have to match to strings. Or have its own bus map and not rely on the one on the `autoprecompiles` crate.

Reply:
String sounds fine to me